CHANGELOG.md

```diff
@@ -13,6 +13,7 @@ When updating the changelog, remember to be very clear about what behavior has c
 and what APIs have changed, if applicable.
 
 ## [Unreleased]
+- Remove unused configurations "dataTemplateGenerator", "restTools", and "avroSchemaGenerator".
```

Comment:
Kinda a nit:
- Since this is backward-compatible and users may (theoretically) need to look at the changelog is something breaks/changes, I'd suggest being more specific here; call these "Gradle configurations" (not just "configurations") added to projects by the pegasus plugin. Remember that the Gradle plugin is just one small fraction of this codebase, so the changelog can be very ambiguous if you're not specific.
- Format this in markdown; wrap the configuration names (and all other code references, for that matter) in backticks.